optimizely/optimizely_user_context.py

```diff
@@ -41,9 +46,37 @@ def __init__(self, optimizely_client, user_id, user_attributes=None):
 
         self._user_attributes = user_attributes.copy() if user_attributes else {}
         self.lock = threading.Lock()
+        self.forced_decisions_map = {}
+        self.log = logger.SimpleLogger(min_level=enums.LogLevels.INFO)
+
+    # decision context
+    class OptimizelyDecisionContext(object):
+        def __init__(self, flag_key, rule_key=None):
+            self.flag_key = flag_key
+            self.rule_key = rule_key
+
+        def __hash__(self):
```

Comment:
I wighted between named tuple and attribute based class.
After a conversation with @jaeopt I opted for the class because it allows better extensibility. it's easy to add a new attribute to OptmizelyDecisionContext should we want to do that in the future.
Writing a class with __eq__ and __hash_- methods is the same as using a named tuple but it offers extensibility. 
__eq__ and __hash_- funcitons are needed to cpmpare items and create a hash. It's what named tuple does under the hood.

true, it looks uglier, but we can save ourselves trouble later.
See more here: https://www.attrs.org/en/stable/why.html#extendability 

I can use namedtuple, but need a reason.